rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoExecutor.java

```diff
@@ -72,11 +91,11 @@ public SQLUndoLog getSqlUndoLog() {
      */
     public void executeOn(Connection conn) throws SQLException {
 
-        // no need undo if the before data snapshot is equivalent to the after data snapshot.
-        if (DataCompareUtils.isRecordsEquals(sqlUndoLog.getBeforeImage(), sqlUndoLog.getAfterImage())) {
+        // when the data validation is not ok 
+        if (!dataValidation(conn)) {
             return;
```

Comment:
should we throw some exception explicitly for the code change later？

Reply:
Sorry, I don’t get your point.

Reply:
I add some log and change the method name for more explicitly.

Reply:
as communicated offline

Reply:
@leizhiyuan   done.